### PR TITLE
added support for unit testing and test results

### PR DESCRIPTION
**Pull request by thescientist13**

### CODE REVIEW

**Comment by obuckley-kenzan** [comment]
Can one of the admins verify this patch?

**Comment by thescientist13** [comment]
@csharon / @jpoo90 
I seem to have an issue that the test-results.xml file is not seen as there by by node, although other files are.  Thoughts?

**Comment by obuckley-kenzan** [comment]
ok to test

**Comment by obuckley-kenzan** [comment]
ok to test

File: test/index.spec.js [hunk 1]
```diff
@@ -1,16 +1,37 @@
 'use strict';
 
-// var nodeTest = require('../');
-// var should = require('chai').should();
+var assert = require('stream-assert');
+var del = require('del');
+var fs = require('fs');
+var gulp = require('gulp');
+//XXX TODO hack to "reset" setting from gulpfile.js so we get generated xml
+var testPipeline = require('../src/index.js')({plugins: {mocha: {reporter: 'spec'}}});
```
Comment (simoami) [inline test/index.spec.js]: The param `{plugins: {mocha: {reporter: 'spec'}}}` overrides your junit reporter in index.js, setting mocha back to a spec reporter which doesn't generate any files.

File: src/index.js [hunk 2]
```diff
@@ -5,8 +5,13 @@ var lazypipe = require('lazypipe');
 var plugins = require('gulp-load-plugins')({lazy: true});
```
Comment (simoami) [inline src/index.js]: `lazy: true` is the default

**Comment by obuckley-kenzan** [comment]
ok to test

**Comment by obuckley-kenzan** [comment]
ok to test

**Comment by obuckley-kenzan** [comment]
Aside from the coverage plugin not working quite right in Jenkins, this is working as expected.  Created a follow task to merge all the results into one directory - issue #5 

cc: @simoami / @jpoo90 / @csharon

**Comment by jpoo90** [comment]
Looks good to me. I tested locally and it is working.

**Comment by obuckley-kenzan** [comment]
move chai to dependencies

**Pull request merged**